backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/cardbutton/BpkCardButton.kt

```diff
@@ -0,0 +1,45 @@
+package net.skyscanner.backpack.compose.cardbutton
+
+import androidx.compose.runtime.Composable
+import net.skyscanner.backpack.compose.cardbutton.internal.BpkSaveCardButtonImpl
+import net.skyscanner.backpack.compose.cardbutton.internal.BpkShareCardButtonImpl
+
+sealed class BpkCardButtonType {
+  data class Save(val checked: Boolean, val onCheckedChange: (Boolean) -> Unit) : BpkCardButtonType()
+  data class Share(val onClick: () -> Unit) : BpkCardButtonType()
+}
+
+enum class BpkCardButtonStyle {
+  Default,
+  Contained,
+  OnDark
+}
+
+enum class BpkCardButtonSize {
+  Default,
+  Small
+}
+
+@Composable
+fun BpkCardButton(
```

Comment:
rather than having a type here it might be better to have two separate composable functions - one `BpkSaveButton` and another `BpkShareButton` and have the different params. that way the components are a little easier to use

Reply:
done